### DISCUSSION TITLE
Binding WireSock network interface in qBittorrent

### DISCUSSION THREAD

**Discussion by RileyXX** [wiresock/WireSockUI#17] [General]
What is the name of the network adapter WireSock uses? I'm trying to bind the network interface in qBittorrent. 

I was doing some testing and I think the network adapter I need to bind is called `Local Area Connection`. Can someone correct me if I'm wrong? I need to make sure I set this up correctly. Thanks.

**Comment by wiresock** [comment]
![image](https://github.com/wiresock/WireSockUI/assets/20592735/5a634da1-aa9e-4803-9986-25ae8903f574)

This depends on the selected mode. In NAT (transparent mode), there isn't a dedicated network interface for the WireGuard tunnel. Traffic is intercepted from the default network connection and encapsulated into the tunnel after NAT is applied. However, in the virtual network adapter mode, a virtual interface named 'Wiresock' is utilized (refer to the attached screenshot).

**Comment by RileyXX** [comment]
Thanks, after checking what you mentioned it looks like I do have the WireSock adapter under my network connections while in the virtual network adapter mode.

<img width="838" alt="07-19-2023_349" src="https://github.com/wiresock/WireSockUI/assets/49823202/25c92acd-caee-490d-83db-aba571fb4ccc">


Problem is in qBittorrent when attempting to bind the network interface to WireSock I don't see any option for "WireSock". From my testing it appears `Local Area Connection` is what is being used for WireSock but I'm not 100% sure if this is correct. When I bind to `Local Area Connection` in qBit and then disconnect from WireSock all connections in qBit stop which seems to be the functionality I'm looking for. When I was setting this up I was expecting the interface in qBit to be called "WireSock". Does this seem like intended behavior or could this be a bug?

<img width="586" alt="07-19-2023_348" src="https://github.com/wiresock/WireSockUI/assets/49823202/fda7b693-be7d-4b38-b205-92afd45d8510">

**Comment by wiresock** [comment]
In a Windows operating system, each network connection is labeled with several names or descriptions, as used throughout the user interface. From the screenshots you've shared, it's evident that the "Local Area Connection" mentioned is in fact referring to the WireSock virtual network adapter. Rest assured, this isn't indicative of any malfunction or bug. Instead, it's part of the normal workings of how Windows manages its network connections.

**Reply by RileyXX** [reply]
I see that makes sense and it's good to know thanks for clarifying on this!
